Project template with conditional content for gitbook vs bs4_book (#1201)
rstudio/bookdown: inst/rstudio/templates/project/resources/common/index.Rmd

--- 
title: "A Minimal Book Example"
author: "John Doe"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
url: https://bookdown.org/johndoe/bookdown-demo

> cover-image: cover.png

Reviewer: We have a copy of the cover image here: https://github.com/rstudio/bookdown/blob/master/inst/examples/images/cover.jpg so I tend not to duplicate it here.

```suggestion
cover-image: https://bookdown.org/yihui/bookdown/images/cover.jpg
```

Reviewer 2: The idea here is to have a template project to start with. 

I am fine with a link here to show that this is possible too, but in our book we usually use a cover image that is in the book, and that gets preprended with the url. See
 * [R Markown Cookbook](https://github.com/yihui/rmarkdown-cookbook/blob/b7d818ccd5ff569194a67d1462a94f85716f5fa8/index.Rmd#L20)
 * [Bookdown Book](https://github.com/rstudio/bookdown/blob/b3f11ba9daed638f3a71e1fecd6c2e626f80d355/inst/examples/index.Rmd#L18)
 * [Definitive Guide](https://github.com/rstudio/rmarkdown-book/blob/ed81b894d5d2665d6ef4657c5840634512b97ff5/index.Rmd#L21)

That is why I thought showing the common practice was a good idea. 

We could maybe not include it and just add a paragraph on how to add a cover image using this field ?

Reviewer: > We could maybe not include it and just add a paragraph on how to add a cover image using this field ?

Yes, that's what I prefer.

Author: Could we re-visit this? Can we make the image super small? This is a tricky one (because you have to use `cover-image` PLUS `url`) for it to work properly, and easier to see with a working example than to read.

Reviewer: Yes, we could use the same approach as https://github.com/rstudio/bookdown/pull/1201#discussion_r675653417.

Reviewer 2: * Retrieve a file from within the package instead of creating a new one.

Reviewer 2: We now retrieve the logo. But this is still not great IMO. I'll explain in main thread

> url: https://bookdown.org/username/bookdown-demo
> cover-image: your-social-sharing-image.png

Reviewer: I feel it would be better to have a working example here like:

```suggestion
# Please update the info below according to the actual url
# and cover-image of your book. If you do not have a cover
# image, you may delete url and cover-image below.
url: https://bookdown.org/yihui/bookdown
cover-image: images/cover.jpg
```

otherwise I prefer commenting out these fields:

```yaml
# url: your book url like https://bookdown.org/yihui/bookdown
# cover-image: path to the cover image like images/cover.jpg
```

I'm okay with either way.

Author: I preferred the second option here, so I made this change.

documentclass: book
bibliography: [book.bib, packages.bib]

> yaml: goes here

Reviewer: What's this?

Reviewer 2: This is a placeholder mechanism so that we can insert the non common part of YAML header for index.Rmd between `gitbook()` and `bs4_book()`. 

My idea was to not duplicate so to have: 

* Common elements and format specific elements
* A mechanism to build the template from those two. 

So this placeholder is replaced in `bookdown_skeleton_insert_yml()` here: https://github.com/rstudio/bookdown/pull/1201/files#diff-f9db8366f31625a76f20758c5adc9d482d7a5243cb2ebe2d5cc2c9cd506fd7ccR61
I have done that to not add a dependency on a templating system like **whisker**

Maybe I should make that more clear by commenting or using a "non supported" YAML syntax ? 

```suggestion
<< placeholder >>
```

Do you have other ideas ? 

Or is aiming at not duplicate the common template project component a wrong idea ? 

The other mechanism is with the comment in Rmd files 
````
<!--bs4_book:start-->

Content specific to bs4_book()

<!--bs4_book:end-->
````

Reviewer: Oh I see. I'm fine with `yaml: goes here`. I thought it was a YAML field that was accidentally left over.

Perhaps a comment like this will be more obvious:

```suggestion
# additional yaml goes here
```

Reviewer 2: I'll make the change to a comment. That will be more explicit for future us and others

Author: @cderv can you resolve this when you make the change you have in mind?

Reviewer 2: Change in 125b862
